sync-core/src/test/java/com/cloudant/sync/datastore/DatastoreManagerTest.java

```diff
@@ -142,28 +151,38 @@ public void createDatastoreWithForwardSlashChar() throws Exception {
 
     @Test
     public void list5Datastores() throws Exception {
+        List<Datastore> opened = new ArrayList<Datastore>();
 
-        for (int i = 0; i < 5; i++) {
-            manager.openDatastore("datastore" + i);
-        }
+        try {
```

Comment:
I don't see how we can nest the try inside the loop because the test needs all 5 datastores to be open before it does its assertions. We also need to ensure we close any already created datastores if we subsequently have a creation failure, so we need to surround the openDatastore with the try anyway. Did you have something specific in mind that I'm missing?